spec/components/notice_spec.rb

```diff
@@ -0,0 +1,56 @@
+require 'rails_helper'
+
+describe "Notice", type: :view do
+  def component_name
+    "notice"
+  end
+
+  it "fails to render when no parameters are given" do
+    assert_empty render_component({})
+  end
+
+  it "renders a notice with only a title and no description" do
+    render_component(title: "Statistics release cancelled")
+    assert_select ".gem-c-notice__title", text: "Statistics release cancelled"
+    assert_select ".gem-c-notice__description", false
+  end
+
+  it "renders a notice with an aria label and region" do
+    render_component(title: "Statistics release cancelled")
+    assert_select ".gem-c-notice[aria-label=Notice][role=region]"
+  end
+
+  it "renders a notice with a title and description text" do
+    render_component(title: "Statistics release cancelled", description_text: "Duplicate, added in error")
+    assert_select ".gem-c-notice__title", text: "Statistics release cancelled"
+    assert_select ".gem-c-notice__description", text: "Duplicate, added in error"
+  end
+
+  # rendering the govspeak component doesn't work here, so we visit a component guide page where it's already rendered and use that instead
+  # this is only a workaround until the govspeak component is also part of the gem
+  # note also that this test will fail if run individually because the govspeak component is rendered differently if run individually
+
+  it "renders a notice with a title and description govspeak" do
+    visit '/component-guide/notice/with_description_govspeak'
+
+    within '.component-guide-preview', match: :first do
+      assert page.has_selector?(".gem-c-notice__title", text: "Statistics release update")
+      assert page.has_selector?(".govuk-govspeak")
```

Comment:
Not sure this assert is needed and the below ones will fail if `.govuk-govspeak` aren't present anyway?

Reply:
Ah yes, that was debug, thanks well spotted.